### ISSUE TITLE
An unexpected error has occured on Ubuntu 22

### ISSUE THREAD

**Issue by hassanrazakhalid** [TiCodeX/SQLSchemaCompare#71]
Hi
I just downloaded version 2024.8.1 and on start up i get crash.

**OS Info:**
PRETTY_NAME="Ubuntu 22.04.5 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.5 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"

**Logs**
```
2024-11-13 11:14:01.664|61813|INFO|electron|Starting SQL Schema Compare v2024.8.1
2024-11-13 11:14:01.966|61813|INFO|electron|Starting service /opt/SQL Schema Compare/bin/TiCodeX.SQLSchemaCompare.UI (25436)
2024-11-13 11:14:48.114|61813|ERROR|electron|Unable to contact service (ERR_CONNECTION_REFUSED)
2024-11-13 11:14:51.275|61813|INFO|electron|Stopping service
2024-11-13 11:14:57.362|62465|INFO|electron|Starting SQL Schema Compare v2024.8.1
2024-11-13 11:14:57.627|62465|INFO|electron|Starting service /opt/SQL Schema Compare/bin/TiCodeX.SQLSchemaCompare.UI (25436)

```

**Comment by drebrez** [comment]
Hi @hassanrazakhalid,

do you have the .NET 8 runtimes installed on your system?

https://learn.microsoft.com/en-us/dotnet/core/install/linux

**Comment by hassanrazakhalid** [comment]
Yes. see result of below command

```
dotnet --info
.NET SDK:
 Version:           8.0.101
 Commit:            6eceda187b
 Workload version:  8.0.100-manifests.69afb982

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  22.04
 OS Platform: Linux
 RID:         linux-x64
 Base Path:   /home/hassan/.dotnet/sdk/8.0.101/

.NET workloads installed:
 Workload version: 8.0.100-manifests.69afb982
There are no installed workloads to display.

Host:
  Version:      8.0.1
  Architecture: x64
  Commit:       bf5e279d92

.NET SDKs installed:
  6.0.418 [/home/hassan/.dotnet/sdk]
  7.0.407 [/home/hassan/.dotnet/sdk]
  8.0.101 [/home/hassan/.dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.26 [/home/hassan/.dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.17 [/home/hassan/.dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.1 [/home/hassan/.dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.26 [/home/hassan/.dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.17 [/home/hassan/.dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.1 [/home/hassan/.dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  DOTNET_ROOT       [/home/hassan/.dotnet]



global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download

```

**Comment by drebrez** [comment]
Hi @hassanrazakhalid,

I just tried with a fresh "Ubuntu 22.04.5" virtual machine and did the following command lines:
```
$ sudo apt install aspnetcore-runtime-8.0
$ wget https://github.com/TiCodeX/SQLSchemaCompare/releases/download/v2024.8.1/SQLSchemaCompare-2024.8.1.deb
$ sudo dpkg -i SQLSchemaCompare-2024.8.1.deb
```
Then I simply clicked the SQL Schema Compare icon in the Applications menu and the application started successfully.

How did you install it on your system? Did you use the dep package or the tar.gz?

Maybe you can try to start it manually with `/opt/SQL\ Schema\ Compare/SQLSchemaCompare` and see if you get some additional log.
The application is composed of 2 parts (frontend and backend), starting the frontend (command above) will start automatically the backend, but maybe you can try to start the backend manually with `/opt/SQL\ Schema\ Compare/bin/TiCodeX.SQLSchemaCompare.UI 25436`
If the backend starts properly you should see "Now listening on https://localhost:25436" and "Application started" messages in the console.

Let me know if you manage to get some additional logs for the manual commands

**Comment by drebrez** [comment]
Hi @hassanrazakhalid,

I've released a new version (2024.12.1) that should fix the problem, could you please try it?

**Comment by drebrez** [comment]
Hi @hassanrazakhalid,

I'm closing the issue as this was probably due the missing .NET Runtime dependency which has been solved in the meantime.

If you still have the problem with the latest version, feel free to re-open it.
